### PR TITLE
PyNumero Documentation

### LINKED ISSUE #1519: Add tests for PyNumero examples

### PR DESCRIPTION
**Pull request by michaelbynum**
## Changes proposed in this PR:
- Add PyNumero documentation for readthedocs
- Add a copy button for example code snippets
- Remove outdated PyNumero examples
- Test PyNumero examples

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by michaelbynum** [comment]
Doctests will probably not pass at first. I'll work on that once I know what is failing.

**Comment by jsiirola** [comment]
What is the relationship between this and the docs in `pyomo/contrib/pynumero/doc/source`?

File: pyomo/contrib/pynumero/examples/tests/test_examples.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+from pyomo.contrib.pynumero.dependencies import numpy_available, scipy_available
+import pyomo.common.unittest as unittest
+
+if not (numpy_available and scipy_available):
+    raise unittest.SkipTest('Pynumero examples needs scipy and numpy')
```
Comment (blnicho) [inline pyomo/contrib/pynumero/examples/tests/test_examples.py:5]: Typo: "examples need"

File: pyomo/contrib/pynumero/examples/tests/test_examples.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    @unittest.skipIf(not ipopt_available,
+                     "feasibility example requires ipopt")
+    def test_feasibility(self):
+        is_feasible = feasibility.main()
+        self.assertTrue(is_feasible)
+
+    @unittest.skipIf(not mumps_available,
+                     'mumps example needs pymumps')
+    def test_mumps_example(self):
+        mumps_example.main()
```
Comment (blnicho) [inline pyomo/contrib/pynumero/examples/tests/test_examples.py:42]: Could you add a few assert statements to these tests?

File: doc/OnlineDocs/conf.py [hunk 3]
```diff
@@ -263,4 +267,20 @@ def setup(app):
     gurobipy_available = True
 except ImportError:
     gurobipy_available = False
+from pyomo.common.dependencies import numpy_available, scipy_available
```
[the remark this answers is not in the document]
Comment (michaelbynum) [reply doc/OnlineDocs/conf.py:270]: Good catch.

File: doc/OnlineDocs/conf.py [hunk 4]
```diff
@@ -263,4 +267,20 @@ def setup(app):
...
+numpy_available = bool(numpy_available)
+scipy_available = bool(scipy_available)
```
[the remark this answers is not in the document]
Comment (michaelbynum) [reply doc/OnlineDocs/conf.py]: I don't think this is needed. I was copying some code above. We will find out for sure with the tests.

File: pyomo/contrib/pynumero/examples/tests/test_examples.py (hunk 1 above)
Comment (michaelbynum) [reply pyomo/contrib/pynumero/examples/tests/test_examples.py]: Fixed. Thanks.

**Review by blnicho — APPROVED** [review]

**Pull request merged**